### PR TITLE
Delay pollers when under heavy load

### PR DESCRIPTION
**Pull request by ssalinas**
Still working out if this is the right strategy. This would allow each poller to determine if it was previously under a heavier than normal load, and then delay it's next run such that other items contending for the scheduler lock (e.g. status updates) have time to get in. Implemented just cleanup poller to start, based on number of cleanup tasks

/cc @wsorenson

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleanupPoller.java [hunk 1]
```diff
@@ -20,21 +21,36 @@
 
   private final SingularityCleaner cleaner;
   private final DisasterManager disasterManager;
+  private final int delayCleanupWhenAboveTasks;
+  private final long delayCleanupForMs;
+
+  private AtomicLong nextRunAfter = new AtomicLong(0);
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleanupPoller.java]: this should be final

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityLeaderOnlyPoller.java [hunk 2]
```diff
@@ -35,6 +36,8 @@
   private SingularityAbort abort;
   private SingularityMesosSchedulerDelegator mesosScheduler;
 
+  private AtomicLong nextRunAfter = new AtomicLong(0);
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityLeaderOnlyPoller.java]: this isn't used

**Comment by ssalinas** [comment]
Sry for unused vars and such, hadn't finished this one yet. I'll clean it up in the morning

**Comment by ssalinas** [comment]
Changed the strategy on this one. Singularity will now keep track of the average status update delta over the last 30s (avg delta is exposed in the state as well). If the delta is over a configured amount (default of 30s), the major scheduler-lock-consuming pollers will all short circuit and offer processing will immediately decline all offers until the delta is back under the configured value. The delta is updated on a 5 second interval by a separate poller.

**Comment by ssalinas** [comment]
merging this into the other leader cache PR

**Pull request merged**